### PR TITLE
Adding an integration test that verifies the heapster metrics endpoint.

### PR DESCRIPTION
**Pull request by vishh**

### CODE REVIEW

**Comment by vishh** [comment]
Ping @rjnagal

**Comment by rjnagal** [comment]
The test failed:

**Comment by rjnagal** [comment]
The test is failing:

+ godep go test -a -v --timeout=30m github.com/GoogleCloudPlatform/heapster/integration/... '--vmodule=*=1' --namespace=default --kube_versions=0.18.2
=== RUN TestHeapster
I0611 23:42:27.122513   18657 framework.go:254] Disabled cluster monitoring
I0611 23:42:27.123087   18657 framework.go:222] checking if existing cluster can be used
I0611 23:42:27.125972   18657 framework.go:225] failed to get kube version info - "the server could not find the requested resource"
I0611 23:42:27.125992   18657 framework.go:261] Setting up new kubernetes cluster version: 0.18.2
I0611 23:42:27.126001   18657 framework.go:153] Bringing down any existing kube cluster
panic: test timed out after 30m0s

goroutine 19 [running]:
testing.func·008()
	/usr/local/go/src/testing/testing.go:681 +0x12f
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x4b

goroutine 1 [chan receive, 27 minutes]:
testing.RunTests(0xba62e0, 0xd86190, 0x1, 0x1, 0xd90501)
	/usr/local/go/src/testing/testing.go:556 +0xad6
testing.(*M).Run(0xc20805a000, 0xd98180)
	/usr/local/go/src/testing/testing.go:485 +0x6c
main.main()
	github.com/GoogleCloudPlatform/heapster/integration/_test/_testmain.go:52 +0x1d5

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0xd8fb20)
	/var/lib/jenkins/workspace/project/src/github.com/GoogleCloudPlatform/heapster/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x78
created by github.com/golang/glog.init·1
	/var/lib/jenkins/workspace/project/src/github.com/GoogleCloudPlatform/heapster/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2a7

goroutine 17 [syscall, 30 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1

File: Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/request.go [hunk 1]
```diff
@@ -20,6 +20,16 @@ import (
 	"bytes"
 	"crypto/tls"
 	"fmt"
+	"io"
+	"io/ioutil"
```
Comment (rjnagal) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/request.go]: Why is it updating a single godep file without any update to Godeps.json?

Comment (vishh) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/request.go]: Removed these changes.

Comment (vishh) [inline Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/request.go]: Unintentional change. Removed.

File: Makefile [hunk 2]
```diff
@@ -18,13 +19,13 @@ sanitize:
...
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go test --test.short github.com/GoogleCloudPlatform/heapster/...
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 godep go test --test.short github.com/GoogleCloudPlatform/heapster/... $(FLAGS)
 
 test-unit-cov: clean deps sanitize build
 	hooks/coverage.sh
 
-test-integration: clean deps
-	godep go test -v --timeout=30m github.com/GoogleCloudPlatform/heapster/integration/... --vmodule=*=2 
+test-integration: clean deps build
+	godep go test -v --timeout=30m github.com/GoogleCloudPlatform/heapster/integration/... --vmodule=*=2 $(FLAGS)
```
Comment (rjnagal) [inline Makefile]: where do we fill the FLAGS? I don't see that anywhere.

Comment (vishh) [inline Makefile]: Its for developer use.

**Comment by rjnagal** [comment]
LGTM

**Comment by rjnagal** [comment]
needs rebase.

**Comment by vishh** [comment]
Rebased @rjnagal

**Comment by vishh** [comment]
@rjnagal: Tests are passing now.

**Pull request merged**